framework/db/BaseActiveRecord.php

```diff
@@ -1682,4 +1707,35 @@ public function offsetUnset($offset)
             unset($this->$offset);
         }
     }
+
+    /**
+     * Resets dependent related models checking if their links contain specific attribute.
+     * @param string $attribute The changed attribute name.
+     */
+    private function resetDependentRelations($attribute)
+    {
+        foreach ($this->_relationsDependencies[$attribute] as $relation) {
+            unset($this->_related[$relation]);
+        }
+        unset($this->_relationsDependencies[$attribute]);
+    }
+
+    /**
+     * Sets relation dependencies for a property
+     * @param string $name property name
+     * @param ActiveQueryInterface $relation relation instance
+     */
+    private function setRelationDependencies($name, $relation)
+    {
+        if (empty($relation->via)) {
+            foreach ($relation->link as $attribute) {
```

Comment:
`link` isn't part of `ActiveQueryInterface`. Probably it's better to hint against `ActiveQuery` in the method signature...